tests/Doctrine/Tests/ORM/Hydration/SimpleObjectHydratorTest.php

```diff
@@ -122,4 +128,36 @@ public function testNullValueShouldNotOverwriteFieldWithSameNameInJoinedInherita
         $result   = $hydrator->hydrateAll($stmt, $rsm);
         $this->assertEquals($result[0], $expectedEntity);
     }
+
+    public function testWrongValuesShouldNotBeConvertedToPhpValue(): void
+    {
+        DBALType::addType(Issue8565EmployeePayloadType::NAME, Issue8565EmployeePayloadType::class);
+        DBALType::addType(Issue8565ManagerPayloadType::NAME, Issue8565ManagerPayloadType::class);
+
+        $rsm = new ResultSetMapping();
+        $rsm->addEntityResult(Issue8565Person::class, 'p');
+        $rsm->addFieldResult('p', 'p__id', 'id');
+        $rsm->addFieldResult('p', 'm__type', 'type', Issue8565Manager::class);
+        $rsm->addFieldResult('p', 'e__type', 'type', Issue8565Employee::class);
+        $rsm->addMetaResult('p', 'discr', 'discr', false, 'string');
+        $rsm->setDiscriminatorColumn('p', 'type');
+        $resultSet = [
+            [
+                'p__id'   => '1',
+                'm__type' => 'type field',
+                'e__type' => 'type field',
```

Comment:
To make sure I understand the PR: here we have a field with an incorrect value in the result set, but since we are going to build a manager object, this field should not even be examined, is that correct?
If yes, why is that field even present in the result set in the first place? And why does it have a value instead of something like `null`? It make no sense for a manager to have the employee `type` field filled in, right?

Reply:
If there is a statement like this:

```sql
SELECT p0.id   AS id_1,
       p0.type AS type_2,
       p0.type AS type_3,
       p0.type AS type_4
FROM person p0
WHERE p0.discr IN ('person', 'manager', 'employee');
```

In result set mapping will appear:

```php
$fieldMappings = [
    'type_2' => 'type',
    'type_3' => 'type',
    'type_4' => 'type',
];
```

So field has a value because there is only **one column** that has **different types for different entities**, and that's why it's not `null`.

Reply:
@tweet9ra please don't mark conversations you did not open as resolved, what if I didn't understand the comment above?
@toshksenof thanks for your comment, I will try to understand it later, right now I'm a bit too tired for that I'm afraid. Or maybe the people I have added as reviewers will beat me to it.

Reply:
After a good night's sleep, I think I finally got it:
Doctrine will select the same column several times, one for each mapping in each subclass.